### PR TITLE
Subkey Updates for v2.0.0-rc6

### LINKED ISSUE #645: Subkey Commands out of date with latest Substrate
Latest Substrate seems to have updated Subkey commands.

https://substrate.dev/docs/en/tutorials/start-a-private-network/keygen

This page says to do:

```
subkey --sr25519 generate
```

where the command is now:

```
subkey generate --scheme sr25519
```

Would be good to double check all other subkey commands too.

### PR DESCRIPTION
**Pull request by danforbes**
Closes #645

### CODE REVIEW

File: docs/knowledgebase/integrate/subkey.md [hunk 1]
```diff
@@ -258,23 +249,21 @@ Recall the SS58 address from the same seed phrase,
...
 Public Key URI `5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd/joe/polkadot/0` is account:
-  Network ID/version: substrate
+  Network ID/version: polkadot
   Public key (hex):   0x06382e496cb8b1664501bbfcc8205c7b1ddc2402e32ef0479b18abdf326ca342
   Account ID:         0x06382e496cb8b1664501bbfcc8205c7b1ddc2402e32ef0479b18abdf326ca342
-  SS58 Address:       5CCrqJffST8uh8RLn2sFJwg1JWs4HkKL88xmdZRdnHt8GDyA
+  SS58 Address:       199ydvjJEQP8fRrjfvFT6WAA8rhz3sUCdhFnrQzLNueSU5G
 ```
 
-Note that the two addresses here match. This is not the case in hard key derivation.
```
Comment (danforbes) [inline docs/knowledgebase/integrate/subkey.md:269]: @shawntabrizi, @joepetrowski - do you  know why this isn't working?

File: docs/knowledgebase/integrate/subkey.md [hunk 2]
```diff
@@ -298,39 +287,38 @@ Secret Key URI `image stomach entry drink rice hen abstract moment nature broken
...
   Account ID:         0xc69291081743ba7e8f587f1e848d1ffc7b634ce21793a231b7ac75d430b59068
-  SS58 Address:       5GZ4srnepXvdsuNVoxCGyVZd8ScDm4gkGLTKuaGARy9akjTa
+  SS58 Address:       15VN2C3igKC7KSP1mbFH7ePmz4bsTNEtLqBp4sFWz4B6wJsa
 ```
 
-Notice that the "SS58-address plus derivation path" produces the same address as the "mnemonic
-phrase plus derivation path plus password." As such, you can reveal your parent address and
-derivation paths without revealing your mnemonic phrase or password, retaining control of all
-derived addresses.
-
```
Comment (danforbes) [inline docs/knowledgebase/integrate/subkey.md:313]: @shawntabrizi, @joepetrowski - also not working :thinking:

File: docs/knowledgebase/integrate/subkey.md (hunk 1 above)
Comment (shawntabrizi) [reply docs/knowledgebase/integrate/subkey.md:269]: There is no `--uri` flag:

```
subkey inspect-key 5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd/joe/polkadot/0 
 
Public Key URI `5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd/joe/polkadot/0` is account:
  Network ID/version: substrate
  Public key (hex):   0x06382e496cb8b1664501bbfcc8205c7b1ddc2402e32ef0479b18abdf326ca342
  Account ID:         0x06382e496cb8b1664501bbfcc8205c7b1ddc2402e32ef0479b18abdf326ca342
  SS58 Address:       5CCrqJffST8uh8RLn2sFJwg1JWs4HkKL88xmdZRdnHt8GDyA
```

Comment (danforbes) [reply docs/knowledgebase/integrate/subkey.md:269]: That does not work for me:
```shell
substrate.dev: subkey --version
subkey 2.0.0-rc6
substrate.dev: subkey inspect-key 5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd/joe/polkadot/0
error: Found argument '5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd/joe/polkadot/0' which wasn't expected, or isn't valid in this context

USAGE:
    subkey inspect-key [FLAGS] [OPTIONS]

For more information try --help
```

Comment (joepetrowski) [reply docs/knowledgebase/integrate/subkey.md:269]: Seems the default changed from Substrate to Polkadot. Matches just fine on the old Subkey:
```
subkey -n polkadot inspect 5CCrqJffST8uh8RLn2sFJwg1JWs4HkKL88xmdZRdnHt8GDyA
Public Key URI `5CCrqJffST8uh8RLn2sFJwg1JWs4HkKL88xmdZRdnHt8GDyA` is account:
  Network ID/version: polkadot
  Public key (hex):   0x06382e496cb8b1664501bbfcc8205c7b1ddc2402e32ef0479b18abdf326ca342
  Account ID:         0x06382e496cb8b1664501bbfcc8205c7b1ddc2402e32ef0479b18abdf326ca342
  SS58 Address:       199ydvjJEQP8fRrjfvFT6WAA8rhz3sUCdhFnrQzLNueSU5G
```

File: docs/knowledgebase/integrate/subkey.md (hunk 2 above)
Comment (joepetrowski) [reply docs/knowledgebase/integrate/subkey.md:313]: Same issue with default network change.

**Review by wheresaddie — CHANGES_REQUESTED** [review]
once --uri flag and default network change is reflected looks good

File: docs/knowledgebase/integrate/subkey.md [hunk 3]
```diff
@@ -140,7 +136,7 @@ The `inspect` command recalculates a key pair's public key and public address gi
 This shows that it is sufficient to back up the seed alone.
 
 ```bash
-$ subkey inspect 0x554b6fc625fbea8f56eb56262d92ccb083fd6eaaf5ee9a966eaab4db2062f4d0
+$ subkey inspect-key --uri 0x554b6fc625fbea8f56eb56262d92ccb083fd6eaaf5ee9a966eaab4db2062f4d0
```
Comment (joepetrowski) [inline docs/knowledgebase/integrate/subkey.md:139]:
```
> subkey inspect-key --uri 0x554b6fc625fbea8f56eb56262d92ccb083fd6eaaf5ee9a966eaab4db2062f4d0
error: Found argument '--uri' which wasn't expected, or isn't valid in this context

USAGE:
    subkey inspect-key [FLAGS] [OPTIONS] [uri]

For more information try --help
```

File: docs/knowledgebase/integrate/subkey.md [hunk 4]
```diff
@@ -151,7 +147,7 @@ Secret Key URI `0x554b6fc625fbea8f56eb56262d92ccb083fd6eaaf5ee9a966eaab4db2062f4
 You can also inspect the key by its mnemonic phrase.
 
 ```bash
-$ subkey inspect "spend report solution aspect tilt omit market cancel what type cave author"
+$ subkey inspect-key --uri "spend report solution aspect tilt omit market cancel what type cave author"
```
Comment (joepetrowski) [inline docs/knowledgebase/integrate/subkey.md:150]: Same. `--uri` is not valid syntax.

File: docs/knowledgebase/integrate/subkey.md [hunk 5]
```diff
@@ -163,32 +159,32 @@ You can inspect password-protected keys either by passing the `--password` flag
...
-Let's say you want to use the same private key on Kusama network. Use `-n` to get your address
-formatted for Kusama. Notice that the public key is the same, but the address has a different
-format.
+Let's say you want to use the same private key on Kusama network. Use `--network` to get your
+address formatted for Kusama. Notice that the public key is the same, but the address has a
+different format.
 
 ```bash
-$ subkey -n kusama inspect "spend report solution aspect tilt omit market cancel what type cave author"
+$ subkey inspect-key --uri "spend report solution aspect tilt omit market cancel what type cave author" --network kusama
```
Comment (joepetrowski) [inline docs/knowledgebase/integrate/subkey.md:182]: Ugh, after the mnemonic works but it's gross.
```suggestion
$ subkey inspect-key --network kusama "spend report solution aspect tilt omit market cancel what type cave author"
```

File: docs/knowledgebase/integrate/subkey.md [hunk 6]
```diff
@@ -163,32 +159,32 @@ You can inspect password-protected keys either by passing the `--password` flag
...
 Secret phrase `spend report solution aspect tilt omit market cancel what type cave author` is account:
   Secret seed:      0x554b6fc625fbea8f56eb56262d92ccb083fd6eaaf5ee9a966eaab4db2062f4d0
   Public key (hex): 0x143fa4ecea108937a2324d36ee4cbce3c6f3a08b0499b276cd7adb7a7631a559
   Account ID:       0x143fa4ecea108937a2324d36ee4cbce3c6f3a08b0499b276cd7adb7a7631a559
-  SS58 Address:     D2sP6XA4DBn3eadsRMYBPoggcDbCuSWUYZ5V63PifURFkcm
+  SS58 Address:     5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd
```
Comment (joepetrowski) [inline docs/knowledgebase/integrate/subkey.md:187]: Nope
```suggestion
  SS58 Address:     D2sP6XA4DBn3eadsRMYBPoggcDbCuSWUYZ5V63PifURFkcm
```

File: docs/knowledgebase/integrate/subkey.md [hunk 7]
```diff
@@ -258,23 +249,21 @@ Recall the SS58 address from the same seed phrase,
 address.
 
 ```bash
-$ subkey inspect "5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd/joe/polkadot/0"
+$ subkey inspect-key --uri "5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd/joe/polkadot/0"
 Public Key URI `5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd/joe/polkadot/0` is account:
-  Network ID/version: substrate
+  Network ID/version: polkadot
```
Comment (joepetrowski) [inline docs/knowledgebase/integrate/subkey.md:254]: How did you get this? I installed the RC6 version and it still defaults to Substrate.

File: docs/knowledgebase/integrate/subkey.md [hunk 8]
```diff
@@ -298,39 +287,38 @@ Secret Key URI `image stomach entry drink rice hen abstract moment nature broken
 
 ```bash
 # SS58-address plus derivation path
-$ subkey inspect "5CZtJLXtVzrBJq1fMWfywDa6XuRwXekGdShPR4b8i9GWSbzB/joe/polkadot/0"
+$ subkey inspect-key --uri "5CZtJLXtVzrBJq1fMWfywDa6XuRwXekGdShPR4b8i9GWSbzB/joe/polkadot/0"
 Public Key URI `5CZtJLXtVzrBJq1fMWfywDa6XuRwXekGdShPR4b8i9GWSbzB/joe/polkadot/0` is account:
-  Network ID/version: substrate
+  Network ID/version: polkadot
```
Comment (joepetrowski) [inline docs/knowledgebase/integrate/subkey.md:292]: substrate

**Review by joepetrowski — COMMENTED** [review]
Need to remove all `--uri`.

File: docs/knowledgebase/integrate/subkey.md (hunk 3 above)
Comment (danforbes) [reply docs/knowledgebase/integrate/subkey.md:139]: Hmmmm not sure what's going on here. This is what I experience:
```shell
~: subkey --version
subkey 2.0.0-rc6
~: subkey inspect-key --uri 0x554b6fc625fbea8f56eb56262d92ccb083fd6eaaf5ee9a966eaab4db2062f4d0
Secret Key URI `0x554b6fc625fbea8f56eb56262d92ccb083fd6eaaf5ee9a966eaab4db2062f4d0` is account:
  Secret seed:      0x554b6fc625fbea8f56eb56262d92ccb083fd6eaaf5ee9a966eaab4db2062f4d0
  Public key (hex): 0x143fa4ecea108937a2324d36ee4cbce3c6f3a08b0499b276cd7adb7a7631a559
  Account ID:       0x143fa4ecea108937a2324d36ee4cbce3c6f3a08b0499b276cd7adb7a7631a559
  SS58 Address:     5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd
```

File: docs/knowledgebase/integrate/subkey.md (hunk 7 above)
Comment (danforbes) [reply docs/knowledgebase/integrate/subkey.md:254]: Here is how I got that:

```shell
~: subkey --version
subkey 2.0.0-rc6
~: subkey inspect-key --uri "5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd/joe/polkadot/0"
Public Key URI `5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd/joe/polkadot/0` is account:
  Network ID/version: polkadot
  Public key (hex):   0x06382e496cb8b1664501bbfcc8205c7b1ddc2402e32ef0479b18abdf326ca342
  Account ID:         0x06382e496cb8b1664501bbfcc8205c7b1ddc2402e32ef0479b18abdf326ca342
  SS58 Address:       199ydvjJEQP8fRrjfvFT6WAA8rhz3sUCdhFnrQzLNueSU5G
```

File: docs/knowledgebase/integrate/subkey.md (hunk 5 above)
Comment (wheresaddie) [reply docs/knowledgebase/integrate/subkey.md:182]: how do you make this sexy and not gross @joepetrowski

Comment (joepetrowski) [reply docs/knowledgebase/integrate/subkey.md:182]: By committing my suggestion :)

File: docs/knowledgebase/integrate/subkey.md (hunk 7 above)
Comment (joepetrowski) [reply docs/knowledgebase/integrate/subkey.md:254]: Well isn't that wonderful
```
~> subkey --version
subkey 2.0.0-rc6
```

File: docs/knowledgebase/integrate/subkey.md [hunk 9]
```diff
@@ -184,7 +184,7 @@ Secret phrase `spend report solution aspect tilt omit market cancel what type ca
   Secret seed:      0x554b6fc625fbea8f56eb56262d92ccb083fd6eaaf5ee9a966eaab4db2062f4d0
   Public key (hex): 0x143fa4ecea108937a2324d36ee4cbce3c6f3a08b0499b276cd7adb7a7631a559
   Account ID:       0x143fa4ecea108937a2324d36ee4cbce3c6f3a08b0499b276cd7adb7a7631a559
-  SS58 Address:     5CXFinBHRrArHzmC6iYVHSSgY1wMQEdL2AiL6RmSEsFvWezd
+  SS58 Address:     D2sP6XA4DBn3eadsRMYBPoggcDbCuSWUYZ5V63PifURFkcm
```
Comment (danforbes) [inline docs/knowledgebase/integrate/subkey.md]: I am still not convinced this is the change we want to make at this time. I think we're still working to pin down the source of the unexpected behavior.

**Comment by danforbes** [comment]
@shawntabrizi, @joepetrowski - I updated the `subkey` doc to specify that it's important to use the tagged 2.0.0-rc6 version. Can you please reevaluate some of the discrepancies we've noticed?

File: docs/knowledgebase/integrate/subkey.md (hunk 9 above)
Comment (joepetrowski) [reply docs/knowledgebase/integrate/subkey.md]: In this case the example is using the `--network kusama` so if the `5CXF...` address actually comes out then this tag is broken and we shouldn't be recommending to install it at all.

File: docs/knowledgebase/integrate/subkey.md [hunk 10]
```diff
@@ -15,13 +15,13 @@ transactions.
...
-you want to play with just Subkey (and not Substrate), you will need to have the Substrate
-dependencies. Use the following two commands to install the dependencies and Subkey, respectively:
+[Substrate](../getting-started/#fast-installation). If you want to play with just Subkey (and not
+Substrate), you will need to have the Substrate dependencies. Use the following two commands to
+install the dependencies and Subkey, respectively:
 
 ```bash
 $ curl https://getsubstrate.io -sSf | bash -s -- --fast
-$ cargo install --force --git https://github.com/paritytech/substrate subkey
+$ cargo install --force subkey --git https://github.com/paritytech/substrate --tag v2.0.0-rc6
```
Comment (riusricardo) [inline docs/knowledgebase/integrate/subkey.md:24]: Does the bug we talked 1 week ago was in `rc5` or `rc6`?
cc: @JoshOrndorff

Comment (JoshOrndorff) [reply docs/knowledgebase/integrate/subkey.md:24]: This is a good change, and will prevent exactly the kind of confusion Dan and Joe experienced above.

I'm not sure what bug you mean though.

**Comment by joepetrowski** [comment]
OK, so `--network` completely does not work on rc6. This is one of the primary use cases of subkey. IMO we should explicitly recommend against this version as it will result in a lot of user confusion.

File: docs/knowledgebase/integrate/subkey.md (hunk 10 above)
Comment (riusricardo) [reply docs/knowledgebase/integrate/subkey.md:24]: Exactly what @joepetrowski commented

**Comment by riusricardo** [comment]
> OK, so `--network` completely does not work on rc6. This is one of the primary use cases of subkey. IMO we should explicitly recommend against this version as it will result in a lot of user confusion.

@JoshOrndorff This is the bug I was talking about. 
@joepetrowski we can use a master commit:
`cargo install --force subkey --git https://github.com/paritytech/substrate --rev 'a7c651c'`

**Comment by joepetrowski** [comment]
> the issue you are getting is in master?

Nope, master works.

**Review by joepetrowski — APPROVED** [review]

**Pull request merged**